src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -190,7 +191,7 @@ public static Course fromUrl(@NotNull URL url, @NotNull ModelFactory modelFactor
       throws IOException, MalformedCourseConfigurationException {
     InputStream inputStream = CoursesClient.fetch(url);
     return Course.fromConfigurationData(
-        new InputStreamReader(inputStream), url.toString(), modelFactory);
+        new InputStreamReader(inputStream, StandardCharsets.UTF_8), url.toString(), modelFactory);
```

Comment:
Hmm... 🤔 Ideally, this kind of manipulation would happen much closer to the DAL but it's not a problem of this PR, _per se_.

Reply:
I guess ideally we should have a method CoursesClient.fetchJSON which would be used for reading all JSONs and this method would already handle UTF-8